src/components/src/test/java/org/apache/jmeter/assertions/jmespath/TestJMESPathAssertion.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.jmeter.assertions.jmespath;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.jmeter.assertions.AssertionResult;
+import org.apache.jmeter.samplers.SampleResult;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Enclosed.class)
+public class TestJMESPathAssertion {
+    @RunWith(Parameterized.class)
+    public static class TestAssertion {
+        private static final String JSON_ARRAY =
+                "{\"people\": [ {\"name\": \"b\", \"age\": 30},"
+                        + " {\"name\": \"a\", \"age\": 50},"
+                        + " {\"name\": \"c\", \"age\": 40}"
+                        + "  ]"
+                        + "}";
+
+        @Parameters
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][] {
+                    // {INVERT, RESPONSE DATA, JMESPATH, VALIDATION, REGEX, EXPECT NULL, RESULT,
+                    // FAILURE, ERROR, FAILURE_MSG}
+                    { Boolean.TRUE, "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]", "[6:6]", Boolean.TRUE, Boolean.FALSE,
+                            Boolean.FALSE, "[]", Boolean.TRUE, Boolean.FALSE, "Value expected not to be equal to []" },
+                    { Boolean.FALSE, "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]", "[6:6]", Boolean.TRUE, Boolean.FALSE,
```

Comment:
It is extremely hard to review and maintain this style of tests.

Could you please add relevant `enum` classes instead of booleans (e.g. the ones that are located inside `TestJMESPathAssertion`)?

Here's an example: https://github.com/apache/jmeter/pull/497/files#diff-52c1ed26a7856184c8efaa2a12116a69R44-R50

Enum like that make test code much simpler to read, then it produce meaningful `toString` when debugging, and it produce meaningful error messages should the test fail.